contracts/src/registry/EnhancedAccessControl.sol

```diff
@@ -0,0 +1,220 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts (last updated v5.0.0) (access/AccessControl.sol)
+
+pragma solidity ^0.8.20;
+
+import {Context} from "@openzeppelin/contracts/utils/Context.sol";
+import {ERC165} from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+
+/**
+ * @dev An enhanced version of OpenZeppelin's AccessControl that allows for:
+ * 
+ * - Context-based roles.
```

Comment:
Rather than calling this 'context' perhaps call it 'principal'?

Reply:
Why? I prefer context as that makes it clearer.

Reply:
"Context" is very generic and collides with the use of it in OZ above. How about "resource"?